packages/tabs/src/Tabs/Tabs.tsx

```diff
@@ -74,62 +67,81 @@ const Tabs = (props: AccessibleTabsProps) => {
   ];
 
-  const childrenArray = useMemo(
-    () => React.Children.toArray(children) as Array<React.ReactElement>,
-    [children],
-  );
+  const accessibilityProps = {
+    ['aria-label']: ariaLabel,
+    ['aria-labelledby']: ariaLabelledby,
+  };
 
-  const handleChange = useCallback(
+  const handleClickTab = useCallback(
     (e: React.SyntheticEvent<Element, MouseEvent>, index: number) => {
       setSelected?.(index);
     },
     [setSelected],
   );
 
-  const getEnabledIndexes: () => [Array<number>, number] = useCallback(() => {
-    const enabledIndexes = childrenArray
-      .filter(child => !child.props.disabled)
-      .map(child => childrenArray.indexOf(child));
-
-    return [enabledIndexes, enabledIndexes.indexOf(selected!)];
-  }, [childrenArray, selected]);
-
-  const handleArrowKeyPress = useCallback(
+  const tabTitleElements = tabDescendants.map(
+    descendant => descendant.element.parentNode as HTMLElement,
+  );
+  const getActiveAndEnabledIndices: () => {
+    activeIndex: number;
+    enabledIndices: Array<number>;
+  } = useCallback(() => {
+    const enabledIndices = tabTitleElements
+      .filter(tabTitleEl => !tabTitleEl.hasAttribute('disabled'))
+      .map(tabTitleEl => tabTitleElements.indexOf(tabTitleEl));
+
+    return {
+      activeIndex: enabledIndices.indexOf(selected),
+      enabledIndices,
+    };
+  }, [selected, tabTitleElements]);
+
+  const handleKeyDown = useCallback(
     (e: KeyboardEvent) => {
-      if (!(e.metaKey || e.ctrlKey)) {
-        if (e.key === keyMap.ArrowRight) {
-          const [enabledIndexes, current] = getEnabledIndexes();
-          setSelected?.(enabledIndexes[(current + 1) % enabledIndexes.length]);
-        } else if (e.key === keyMap.ArrowLeft) {
-          const [enabledIndexes, current] = getEnabledIndexes();
-          setSelected?.(
-            enabledIndexes[
-              (current - 1 + enabledIndexes.length) % enabledIndexes.length
-            ],
-          );
-        }
-      }
+      if (e.metaKey || e.ctrlKey) return;
+
+      if (e.key !== keyMap.ArrowRight && e.key !== keyMap.ArrowLeft) return;
+
+      const { activeIndex, enabledIndices } = getActiveAndEnabledIndices();
+      const numberOfEnabledTabs = enabledIndices.length;
+      const indexToUpdateTo =
+        enabledIndices[
+          (e.key === keyMap.ArrowRight
+            ? activeIndex + 1
+            : activeIndex - 1 + numberOfEnabledTabs) % numberOfEnabledTabs
+        ];
+      setSelected?.(indexToUpdateTo);
+      tabTitleElements[indexToUpdateTo].focus();
     },
-    [getEnabledIndexes, setSelected],
+    [getActiveAndEnabledIndices, setSelected, tabTitleElements],
   );
 
+  useEffect(() => {
+    if (
+      !isControlled ||
+      tabTitleElements.length === 0 ||
+      tabTitleElements[controlledSelected] === undefined
+    ) {
+      return;
+    }
+
+    tabTitleElements[controlledSelected].focus();
+  }, [controlledSelected, isControlled, tabTitleElements]);
```

Comment:
This will technically fire if/when `tabTitleElements` or `isControlled` changes. 

I can imagine a case where some external event causes tabs to be added/removed, which would trigger this effect

Reply:
If I comment out this `useEffect` locally, changing `selected` in Storybook does not update the focus (only the green "selected" indicator)

Reply:
@TheSonOfThomp to avoid having the effect run based off external events when tabs are added/removed, I opted to leave out the `useEffect`. When testing on the `Controlled` story instance, focus seems to update as intended. Let me know if you think otherwise, and I can add it back in